File: mobile-frontend/static/tabbar/generate-icons.py
"""
扫码图标生成脚本
运行方法：python generate-icons.py
需要安装：pip install Pillow
"""

from PIL import Image, ImageDraw
import os

def draw_qr_code_icon(draw, color, size, line_width):
    """绘制扫码图标"""
    padding = int(size * 0.2)
    corner_size = int(size * 0.25)

    # 左上角
    draw.line([(padding, padding + corner_size), (padding, padding), (padding + corner_size, padding)],
              fill=color, width=line_width)

    # 右上角
    draw.line([(size - padding - corner_size, padding), (size - padding, padding), (size - padding, padding + corner_size)],
              fill=color, width=line_width)

    # 左下角
    draw.line([(padding, size - padding - corner_size), (padding, size - padding), (padding + corner_size, size - padding)],
              fill=color, width=line_width)

    # 右下角
    draw.line([(size - padding - corner_size, size - padding), (size - padding, size - padding),
               (size - padding, size - padding - corner_size)],
              fill=color, width=line_width)

    # 中间扫描线（水平）
    draw.line([(padding + corner_size, size // 2), (size - padding - corner_size, size // 2)],
              fill=color, width=line_width)

    # 中间扫描线（垂直）
    draw.line([(size // 2, padding + corner_size), (size // 2, size - padding - corner_size)],
              fill=color, width=line_width)

def generate_icons():
    size = 81
    line_width = 2

    # 生成灰色图标
    img_gray = Image.new('RGBA', (size, size), (255, 255, 255, 0))
    draw_gray = ImageDraw.Draw(img_gray)
    draw_qr_code_icon(draw_gray, (122, 174, 131), size, line_width)  # #7A7E83
    img_gray.save('scan.png')
    print('[OK] Generated scan.png')

    # 生成蓝色图标
    img_blue = Image.new('RGBA', (size, size), (255, 255, 255, 0))
    draw_blue = ImageDraw.Draw(img_blue)
    draw_qr_code_icon(draw_blue, (14, 165, 233), size, line_width)  # #0ea5e9
    img_blue.save('scan-active.png')
    print('[OK] Generated scan-active.png')

    print('\n[SUCCESS] Icons generated successfully!')
    print(f'[PATH] {os.path.dirname(os.path.abspath(__file__))}')

if __name__ == '__main__':
    try:
        generate_icons()
    except ImportError:
        print('[ERROR] Pillow library not found')
        print('Please run: pip install Pillow')

File: mobile-frontend/static/tabbar/generate-icons.js
/**
 * 扫码图标生成脚本
 * 运行方法：node generate-icons.js
 * 需要安装：npm install canvas
 */

const { createCanvas } = require('canvas');

function drawQRCodeIcon(ctx, color, size, lineWidth) {
    ctx.strokeStyle = color;
    ctx.lineWidth = lineWidth;
    ctx.lineCap = 'round';

    const padding = size * 0.2;
    const cornerSize = size * 0.25;

    // 左上角
    ctx.beginPath();
    ctx.moveTo(padding, padding + cornerSize);
    ctx.lineTo(padding, padding);
    ctx.lineTo(padding + cornerSize, padding);
    ctx.stroke();

    // 右上角
    ctx.beginPath();
    ctx.moveTo(size - padding - cornerSize, padding);
    ctx.lineTo(size - padding, padding);
    ctx.lineTo(size - padding, padding + cornerSize);
    ctx.stroke();

    // 左下角
    ctx.beginPath();
    ctx.moveTo(padding, size - padding - cornerSize);
    ctx.lineTo(padding, size - padding);
    ctx.lineTo(padding + cornerSize, size - padding);
    ctx.stroke();

    // 右下角
    ctx.beginPath();
    ctx.moveTo(size - padding - cornerSize, size - padding);
    ctx.lineTo(size - padding, size - padding);
    ctx.lineTo(size - padding, size - padding - cornerSize);
    ctx.stroke();

    // 中间扫描线（水平）
    ctx.beginPath();
    ctx.moveTo(padding + cornerSize, size / 2);
    ctx.lineTo(size - padding - cornerSize, size / 2);
    ctx.stroke();

    // 中间扫描线（垂直）
    ctx.beginPath();
    ctx.moveTo(size / 2, padding + cornerSize);
    ctx.lineTo(size / 2, size - padding - cornerSize);
    ctx.stroke();
}

function generateIcons() {
    const size = 81;

    // 生成灰色图标
    const canvasGray = createCanvas(size, size);
    const ctxGray = canvasGray.getContext('2d');
    drawQRCodeIcon(ctxGray, '#7A7E83', size, 2);

    const fs = require('fs');
    const bufferGray = canvasGray.toBuffer('image/png');
    fs.writeFileSync('scan.png', bufferGray);
    console.log('✅ 已生成 scan.png');

    // 生成蓝色图标
    const canvasBlue = createCanvas(size, size);
    const ctxBlue = canvasBlue.getContext('2d');
    drawQRCodeIcon(ctxBlue, '#0ea5e9', size, 2);

    const bufferBlue = canvasBlue.toBuffer('image/png');
    fs.writeFileSync('scan-active.png', bufferBlue);
    console.log('✅ 已生成 scan-active.png');

    console.log('\n🎉 图标生成完成！');
    console.log('📁 文件位置：' + __dirname);
}

generateIcons();
